lightshow/postprocess/parse.py

```diff
@@ -0,0 +1,361 @@
+                }
+            else: 
+                dict_output[element][absorber]['spectrum'][:,1] += np.loadtxt( sub_path, usecols=(2)  )
+                dict_output[element][absorber]['num_polar'] += 1
+
+    # Raise error if there is no ``absspct...`` file in the directory.                     
+    try: 
+        assert num_spectra > 0, "\'absspct...\' file not found in %s"%path 
+    except AssertionError as msg: 
+        print(msg)
+        return
+    
+    for element in dict_output.keys(): 
+        for absorber in dict_output[element].keys():
+            dict_output[element][absorber]['spectrum'][:,1] /= dict_output[element][absorber]['num_polar']
+            del dict_output[element][absorber]['num_polar']
+
+    # Try to extract Fermi energy from ``scf.out`` file. 
+    file_out = scf_out_file
+    try: 
+        with open(file_out, 'r') as f:
+            for line in f.readlines():
+                if 'Fermi energy' in line:
+                    efermi = float(line.split()[4])
+                if '!' in line:
+                    total_energy = float(line.split()[-2]) * 13.605703976
+        try: 
+            dict_output['efermi'] = efermi
+        except NameError as err: 
+            print('Cannot find Fermi energy in', file_out)
+        try: 
+            dict_output['total_energy'] = total_energy
+        except: 
+            print('Cannot find total energy in', file_out) 
+    except OSError: 
+        print(file_out, 'not found. Fermi energy and total energy not extracted.')
+    
+    return dict_output
+
+
+def extract_exciting(path, INFO_out_file=None): 
+    """Extract spectrum from exciting output directory. 
+    
+    Parameters
+    ----------
+    path : str
+        exciting spectra output directory. 
+    INFO_path: str, optional
+        exciting output directory where ``INFO.OUT`` file is stored. 
+        If INFO_path is not specified, will use the spectra path. 
+        
+    Returns
+    -------
+    dict_output
+        A dictionary containing the spectrum. The spectrum is averaged from all polarization directions in ``path``.
+        Fermi level and total energy are extracted if ``INFO.OUT`` file is provided, with unit eV.
+        ``{"spectrum": [..., ...], "efermi": ..., "total_energy": ...}``
+
+    Raises
+    ------
+    AssertionError
+        If the output path does not contain any file that starts with ``EPSILON`` and ends with ``OUT``. 
+    """
+    
+    dict_output = {}
+    
+    path = Path(path)
+    if INFO_out_file is None:
+        INFO_out_file = path / "INFO.OUT"
+    
+    num_polar = 0
+    spectra = None
+    for sub_path in path.rglob('EPSILON*OUT'): 
+        num_polar += 1
+        if spectra is None: 
+            spectra = np.loadtxt(sub_path, usecols=(0,2))
+        else: 
+            spectra[:,1] += np.loadtxt( sub_path, usecols=(2)  )
+    ### Raise error if there is no ``EPSILON...OUT`` file in the directory.                     
+    try: 
```

Comment:
As Matt already commented, this try/except/assert blocks are not very helpful. Simply raise the error

Reply:
Resolved in the updated code on 12.3.2023